benchexec/tools/btormc.py

```diff
@@ -0,0 +1,47 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.result as result
+import benchexec.tools.template
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Tool info for BtorMC -- A Bounded Model Checker for Btor2
+    """
+
+    REQUIRED_PATHS = ["build/"]
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("btormc", subdir="build/bin")
+
+    def name(self):
+        return "BtorMC"
+
+    def project_url(self):
+        return "https://github.com/Boolector/boolector"
+
+    def version(self, executable):
+        return self._version_from_tool(executable, stdin="")
+
+    def cmdline(self, executable, options, task, rlimits):
+        return [executable] + options + [task.single_input_file]
+
+    def determine_result(self, run):
+        """
+        @return: status of BtorMC after executing a run
+        """
+        if run.was_timeout:
+            return result.RESULT_TIMEOUT
```

Comment:
Note that it is only necessary to do this if you want to make sure that the below code is not executed in timeout cases. This could be the case if the tool always prints out lines matching the patterns below, but they are meaningless in timeouts (though then you probably want to broaden this check to `was_terminated`).

It is not the responsibility of the tool-info module to check for timeouts.

Reply:
Addressed in d90d0fe